sdk/samples/keyvault/keyvault/samples/src/keys_client_example.c

```diff
@@ -57,6 +58,17 @@
 #define CLIENT_SECRET_ENV "AZURE_CLIENT_SECRET"
 #define URI_ENV "AZURE_KEYVAULT_URL"
 
+#ifdef TRANSPORT_CURL
+#define EXIT(code) \
+  do \
+  { \
+    curl_global_cleanup(); \
```

Comment:
You could write a callback function and pass that to atexit when main initializes: https://www.tutorialspoint.com/c_standard_library/c_function_atexit.htm.
Then, we don't need more macros which obscure what the code is really doing and we'd like customers to NOT use our sample macros in their production code.

But, the use of atexit is something I'd be happy to encourage customers to use.

Reply:
yes!, this is great! thank you.
About the Macros, IMHO, we should hide all code that we believe not relevant for the sample behind Macros.  Then just show the main idea of the sample.

I find validating each function result value as a distraction since it is all duplicated code that ends up taking most of the visibility from sample on a glance.

Reply:
Microsoft has a history of samples that do not show error handling and then customers copy & paste this code and do not add the error handling. So, our team has decided that error handling in samples is first class and what customers have to do in real life. 

Then, I think it's best to avoid macros in samples (as much as possible) because they insert a level of indirection. That is, the customer MUST learn what the macro does in order to understand what the sample is really doing and this is the opposite of what a sample should be about. In fact, this is one of the reasons why all modern languages (like C#, Go, etc.) have no concept of macros at all.